File: packages/ts/src/strategies/fix-commas.ts
/**
 * fix-commas: Remove trailing commas before } and ].
 * Must-isolate: applied before fix-keys. Runs before fix-keys.
 */
export const fixCommas = (text: string): string => {
  // Remove trailing comma before } or ] accounting for optional whitespace/newlines
  return text.replace(/,(\s*[}\]])/g, "$1");
};

File: packages/python/src/output_guard/strategies/fix_commas.py
"""fix-commas: Remove trailing commas before } and ].

Mirror of packages/ts/src/strategies/fix-commas.ts.
"""

from __future__ import annotations

import re

_TRAILING = re.compile(r",(\s*[}\]])")


def fix_commas(text: str) -> str:
    return _TRAILING.sub(r"\1", text)
